airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/AirbyteConnectorRunnable.kt

```diff
@@ -40,8 +42,19 @@ class AirbyteConnectorRunnable : Runnable {
                     "Failed ${operation::class} operation execution."
                 }
             }
-            outputConsumer.accept(exceptionHandler.handle(e))
-            throw e
+            if (operationName == "check") {
+                // During check, we don't fail the command on uncaught error. We assume the check as
+                // failed and return a trace + a connection status message.
+                val exception: Throwable? =
+                    if (e.message == "Failed to initialize connector operation") e.cause else e
+                val (errorTraceMessage, connectionStatusMessage) =
+                    exceptionHandler.handleCheckFailure(exception ?: e)
+                outputConsumer.accept(errorTraceMessage)
+                outputConsumer.accept(connectionStatusMessage)
+            } else {
+                outputConsumer.accept(exceptionHandler.handle(e))
+                throw e
+            }
```

Comment:
Are there unit tests for this?

Nit: I'd break this out as a stand alone method (or what have you) for unit testing.